File: algorithms/easy/n_repeated_element_in_size_2n_array/solution.py
class Solution:
    def repeatedNTimes(self, nums: list) -> int:
        num_dict = dict()
        for idx, num in enumerate(nums):
            if num in num_dict:
                return num
            
            if num not in num_dict:
                num_dict[num] = True

solution = Solution()
print(solution.repeatedNTimes([1, 2, 3, 3]) == 3)
print(solution.repeatedNTimes([2, 1, 2, 5, 3, 2]) == 2)
print(solution.repeatedNTimes([5, 1, 5, 2, 5, 3, 5, 4]) == 5)

File: algorithms/easy/n_repeated_element_in_size_2n_array/solution.go
package main

import "fmt"

func repeatedNTimes(nums []int) int {
	numMap := make(map[int]bool, len(nums))
	for _, num := range nums {
		if numMap[num] {
			return num
		}

		numMap[num] = true
	}

	return 0
}

func main() {
	fmt.Println(repeatedNTimes([]int{1, 2, 3, 3}) == 3)
	fmt.Println(repeatedNTimes([]int{2, 1, 2, 5, 3, 2}) == 2)
	fmt.Println(repeatedNTimes([]int{5, 1, 5, 2, 5, 3, 5, 4}) == 5)
}
